### PR TITLE
Fix ChannelListWindow

### PR DESCRIPTION
**Pull request by GuskoMaksim**
Changing the logic of adding channels

### CODE REVIEW

**Comment by otratsiuk** [comment]
Could you please link the issue to this pull request

**Review by otratsiuk — APPROVED** [review]

File: Client/Client.Qt/Widgets/ChannelListWindow.hpp [hunk 1]
```diff
@@ -21,22 +21,33 @@ class ChannelListWindow : public QWidget
...
-    inline static std::vector<std::string> _channelNames{};
-    std::unique_ptr<ListWidget>            _channelList;
-    std::unique_ptr<FlatButton>            _addChannelButton;
-    std::unique_ptr<FlatButton>            _updateChannelButton;
-    std::unique_ptr<QVBoxLayout>           _vBoxLayout;
-    std::shared_ptr<ListWidget>            _channelListMainWindow;
+    struct channelStruct
+    {
+        uint16_t _channelId;
+        bool _addToMainChannelsWindow = false;
```
Comment (StasDash) [inline Client/Client.Qt/Widgets/ChannelListWindow.hpp:45]: _ is for private members

File: Client/Client.Qt/Widgets/ChannelListWindow.hpp [hunk 2]
```diff
@@ -21,22 +21,33 @@ class ChannelListWindow : public QWidget
...
     void updateChannelListWindow();
 
 private:
-    inline static std::vector<std::string> _channelNames{};
-    std::unique_ptr<ListWidget>            _channelList;
-    std::unique_ptr<FlatButton>            _addChannelButton;
-    std::unique_ptr<FlatButton>            _updateChannelButton;
-    std::unique_ptr<QVBoxLayout>           _vBoxLayout;
-    std::shared_ptr<ListWidget>            _channelListMainWindow;
+    struct channelStruct
```
Comment (fisyamaloy) [inline Client/Client.Qt/Widgets/ChannelListWindow.hpp:42]: Naming of struct & classes - camel case with first big letter.

**Review by StasDash — APPROVED** [review]

File: Client/Client.Qt/Widgets/ChannelListWindow.hpp (hunk 1 above)
Comment (StasDash) [reply Client/Client.Qt/Widgets/ChannelListWindow.hpp:45]: _

**Review by StasDash — CHANGES_REQUESTED** [review]

File: Client/Client.Qt/Widgets/ChannelListWindow.cpp [hunk 3]
```diff
@@ -44,37 +44,13 @@ void ChannelListWindow::updateChannelList()
...
+            for (auto &channel : _channelsMap)
             {
-                if (_channelList->count() == 0)
+                if (!channel.second._addToMainChannelsWindow)
                 {
-                    _channelList->addItem(QString::fromStdString(*it));
+                    _channelList->addItem(QString::fromStdString(channel.first));
+                    channel.second._addToMainChannelsWindow = true;
                 }
-                for (int i = 0; i < _channelList->count(); ++i)
```
Comment (StasDash) [inline Client/Client.Qt/Widgets/ChannelListWindow.cpp:66]: Does this break https://github.com/f1nal3/Juniorgram/issues/217?

Comment (GuskoMaksim) [reply Client/Client.Qt/Widgets/ChannelListWindow.cpp:66]: No

File: Client/Client.Qt/Widgets/ChannelListWindow.hpp [hunk 4]
```diff
@@ -21,22 +21,33 @@ class ChannelListWindow : public QWidget
...
-    inline static std::vector<std::string> _channelNames{};
-    std::unique_ptr<ListWidget>            _channelList;
-    std::unique_ptr<FlatButton>            _addChannelButton;
-    std::unique_ptr<FlatButton>            _updateChannelButton;
-    std::unique_ptr<QVBoxLayout>           _vBoxLayout;
-    std::shared_ptr<ListWidget>            _channelListMainWindow;
+    struct ChannelStruct
+    {
+        uint16_t channelId;
+        bool     addToMainChannelsWindow = false;
```
Comment (fisyamaloy) [inline Client/Client.Qt/Widgets/ChannelListWindow.hpp:45]: why do u need this field?

File: Client/Client.Qt/Widgets/ChannelListWindow.cpp [hunk 5]
```diff
@@ -105,10 +81,17 @@ void ChannelListWindow::addChannelToMainChannelWidget()
             _channelListMainWindow->addItem(_channelList->takeItem(_channelList->currentRow()));
         }
     }
-    this->hide();
 }
 
-void ChannelListWindow::addChannelInfo(const std::string& nameOfChannels) { _channelNames.push_back(nameOfChannels); }
+void ChannelListWindow::addChannelInfoMap(const std::map<std::string, uint16_t> &nameOfChannelsMap)
+{
+    for (auto channel : nameOfChannelsMap)
```
Comment (fisyamaloy) [inline Client/Client.Qt/Widgets/ChannelListWindow.cpp:88]: don't create every time copy. The better way is using & or &&.

File: Client/Client.Qt/Widgets/ChannelListWindow.hpp (hunk 4 above)
Comment (fisyamaloy) [reply Client/Client.Qt/Widgets/ChannelListWindow.hpp:45]: if we want addiing object with the same key, it will not add.

**Comment by GuskoMaksim** [comment]
I am waiting for changes from the "Fix message string" branch

**Review by vDrWool — APPROVED** [review]

**Review by Svidunovich — APPROVED** [review]

**Pull request closed without merging**